File: arranging_cats/arrange.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{
    int t;
    cin >> t;
    for (int i = 0; i < t; i++)
    {
        int n;
        cin >> n;
        string s, f;
        cin >> s >> f;
        int ntr = 0;
        int nta = 0;
        for (int j = 0; j < n; j++)
        {
            if (s.at(j) == '1' && f.at(j) == '0')
            {
                ntr++;
            }

            else if (s.at(j) == '0' && f.at(j) == '1')
            {
                nta++;
            }
        }

        int result = (ntr > nta) ? ntr : nta;
        cout << result << "\n";

    }
}

File: arranging_cats/arrange.py
def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        s = input()
        f = input()
        nta = 0
        ntr = 0
        for i in range(n):
            if s[i] == "1" and f[i] == "0":
                ntr += 1
            elif s[i] == "0" and f[i] == "1":
                nta += 1
        print(ntr if ntr > nta else nta)
        
    
if __name__ == "__main__":
    main()
